lib/dao.js

```diff
@@ -424,13 +435,26 @@ DataAccessObject.updateOrCreate = DataAccessObject.upsert = function upsert(data
 
         var connector = self.getConnector();
 
+        context = {
+          Model: Model,
+          where: ctx.where,
+          data: removeUndefined(data),
+          currentInstance: inst,
+          hookState: ctx.hookState,
+          options: options
+        };
+
         if (Model.settings.validateUpsert === false) {
-          update = removeUndefined(update);
-          if (connector.updateOrCreate.length === 4) {
-            connector.updateOrCreate(Model.modelName, update, options, done);
-          } else {
-            connector.updateOrCreate(Model.modelName, update, done);
-          }
+          Model.notifyObserversOf('persist', context, function(err) {
+            if (err) return cb(err);
```

Comment:
There is a subtle difference between calling `cb(err)` and `done(err)`. I think we should call `done`.